sqldelight-compiler/integration-tests/src/test/kotlin/com/squareup/sqldelight/core/integration/IntegrationTest.kt

```diff
@@ -102,6 +103,15 @@ class IntegrationTest {
         |WHERE inner_type = ?;
         |""".trimMargin(), temporaryFolder, "Team.sq")
 
+    FixtureCompiler.writeSql("""
+        |CREATE TABLE `group` (`index` INTEGER PRIMARY KEY NOT NULL);
+        |
+        |INSERT INTO `group` VALUES (1), (2), (3);
+        |
+        |selectAll:
+        |SELECT `index` FROM `group`;
```

Comment:
I'm not sure this is the right place for this test. @AlecStrong?

We're also missing coverage of single quote, double quote, and square bracket use.

Reply:
i dont mind having an integration test for it to verify that the normalizing doesnt effect the results at all.

would love a test in `tables/InterfaceGeneration` similar to https://github.com/square/sqldelight/blob/master/sqldelight-compiler/src/test/kotlin/com/squareup/sqldelight/core/tables/InterfaceGeneration.kt#L110-L166

for the other cases you mentioned

Reply:
I added a case for testing all variants. I tests all version of escaped column names but only one of the versions for table names. I could add one per version if you'd like.